### PR TITLE
Prefix and MaxLine writers

### PR DESCRIPTION
**Pull request by willhite**
Cleanup Prefix and MaxLine writers.

These were previously one writer that was really only usable by
noms. This commit separates them into to separate writers that
can be used independently or combined. The also moved into
go/utils/writers so they can be used by other code.

The main impetus to do this was to fix Bug #2593.

### CODE REVIEW

**Comment by arv** [comment]
Can you expand the commit message to explain what this is about?

File: go/util/writers/max_line_writer.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package writers
+
+import "io"
+
+var (
+	MaxLinesErr = MaxLinesError{"Maximum number of lines written"}
```
Comment (arv) [inline go/util/writers/max_line_writer.go]: Godoc

File: go/util/writers/max_line_writer.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+)
+
+type MaxLinesError struct {
```
Comment (arv) [inline go/util/writers/max_line_writer.go]: Godoc

File: go/util/writers/max_line_writer.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	msg string
+}
+
+func (e MaxLinesError) Error() string { return e.msg }
```
Comment (arv) [inline go/util/writers/max_line_writer.go]: godoc ...

File: go/util/writers/prefix_writer.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package writers
+
+import "io"
+
+type PrefixWriter struct {
```
Comment (arv) [inline go/util/writers/prefix_writer.go]: godoc

**Comment by willhite** [comment]
@arv PTAL

**Comment by willhite** [comment]
@arv PTAL. Also, I found another instance of a prefixWriter in noms diff that I was able to remove and replace with this new implementation.

File: go/types/encode_human_readable.go [hunk 5]
```diff
@@ -327,13 +339,24 @@ func EncodedValue(v Value) string {
 	return encodedValueFormat(v, 'g')
 }
 
+func EncodedValueMaxLines(v Value, maxLines uint32) string {
```
Comment (arv) [inline go/types/encode_human_readable.go]: godoc here too

File: go/util/writers/prefix_writer.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package writers
+
+import "io"
+
+// PrefixWriter makes it easy to prefix newlines with a custom prefix. Each time
```
Comment (arv) [inline go/util/writers/prefix_writer.go]:
```
Each time it writes a byte after a newline ('\n') ...
```

**Comment by willhite** [comment]
@arv PTAL

File: go/types/encode_human_readable.go [hunk 7]
```diff
@@ -335,10 +335,13 @@ func EncodedIndexValue(v Value) string {
 	return encodedValueFormat(v, 'f')
 }
 
+// EncodedValue returns a string containing the serialization of a value.
```
Comment (arv) [inline go/types/encode_human_readable.go]: Maybe serialization -> human readable serialization

**Pull request merged**